src/history_function.jl

```diff
@@ -31,16 +33,21 @@ function (f::HistoryFunction)(p, t, ::Type{Val{deriv}}=Val{0}; idxs=nothing) whe
     else
       return f.h(p, t, Val{deriv}; idxs = idxs)
     end
-  elseif integrator.tdir * t <= integrator.tdir * sol.t[end] # Put equals back
+  end
+  
+  tdir_solt = tdir * sol.t[end]
+  if tdir_t <= tdir_solt
     return sol.interp(t, idxs, Val{deriv}, p)
   end
 
   # history function is evaluated at time point past the final time point of
   # the current solution
-  f.isout = true
+  if tdir_t - tdir_solt > 10 * eps(tdir_t)
```

Comment:
I noticed that even for constrained time stepping due to floating point issues the solution is sometimes extrapolated. However, in that case we do not want to perform any fixed-point iterations. So basically there are two options: 1) check if the algorithm is constrained (which is a bit tricky since the DDEIntegrator does not contain any reference to the MethodOfSteps algorithm) and in that case do not check if the history function was extrapolated, or 2) generally do not perform fixed-point iterations if the extrapolation is very close to the final time point of the history and throw an error if still fixed-point iterations are performed in the constrained case. I went for the second alternative.

Reply:
how would floating point issues cause accidental extrapolation?

Reply:
The delays are subtracted from the current time point, and thus `t - tau` can be greater than `tprev` even if `t = tprev + tau` (happens in the constrained tests, e.g.).

Reply:
oh yes, that's terrifying. This is a good workaround.